account-model/src/account/account_builder.rs

```diff
@@ -0,0 +1,321 @@
+use iso_currency::Currency;
+use isocountry::CountryCode;
+use log::debug;
+use uuid::Uuid;
+
+use crate::Account;
+use crate::account::{Classification, KnownBankIdCode};
+use crate::account::Classification::{BUSINESS, PERSONAL};
+
+#[derive(Debug)]
+pub struct Builder {
+    id: Option<Uuid>,
+    organisation_id: Option<Uuid>,
+    country: Option<CountryCode>,
+    currency: Option<Currency>,
+    bank_id_code: Option<String>,
+    bank_id: Option<String>,
+    bic: Option<String>,
+    number: Option<String>,
+    iban: Option<String>,
+    title: Option<String>,
+    classification: Classification,
+}
+
+impl Builder {
+    pub fn new_account() -> Builder {
+        Builder {
+            id: None,
+            organisation_id: None,
+            country: None,
+            currency: None,
+            bank_id_code: None,
+            bank_id: None,
+            bic: None,
+            number: None,
+            iban: None,
+            title: None,
+            classification: Classification::PERSONAL,
+        }
+    }
+
+    pub fn with_account_id(&mut self, id: Uuid) -> &mut Builder {
```

Comment:
`pub fn with_account_id(&mut self, id: Uuid) -> &mut Builder` trochę nie rozumiem co by się stało gdybym zadeklarował jako 
`pub fn with_account_id(self, id: Uuid) -> Builder` 
czy po wyjściu z tej funkcji nastąpiłby move zmiennej builder-a, a więc i kopia wszystkiego co tam się znajdowało ?

Reply:
mogloby sie tak zdarzyc

Reply:
czyli metoda obiektu (jeśli to można tak nazwać) nie przynajleży do jakiejś konkretnej instancji tylko instancja jest przekazywana z zadeklarowanym modyfikatorem do tkaiej funkcji i tylko instancje danego typu mają do niej dostęp